flatpak-import.py

```diff
@@ -0,0 +1,212 @@
+                         help='verbose log output')
+    aparser.add_argument('-d', '--debug', action='store_true',
+                         help='debug log output')
+
+    config = ConfigParser()
+    config.read('flatpak-import.conf')
+    aparser.set_defaults(**dict(config.items('defaults')))
+
+    args = aparser.parse_args()
+
+    if args.debug:
+        logging.basicConfig(level=logging.DEBUG)
+    elif args.verbose:
+        logging.basicConfig(level=logging.INFO)
+    else:
+        logging.basicConfig(level=logging.WARNING)
+
+
+    ### mmap the flatpak file and create a GLib.Variant from it
+
+    mapped_file = GLib.MappedFile.new(args.flatpak, False)
+    mapped_bytes = mapped_file.get_bytes()
+    ostree_static_delta_superblock_format = GLib.VariantType(
+            OSTREE_STATIC_DELTA_SUPERBLOCK_FORMAT)
+    delta = GLib.Variant.new_from_bytes(
+            ostree_static_delta_superblock_format,
+            mapped_bytes,
+            False)
+
+
+    ### parse flatpak metadata
+
+    # use get_child_value instead of array index to avoid
+    # slowdown (constructing the whole array?)
+    checksum_variant = delta.get_child_value(3)
+    if not OSTree.validate_structureof_csum_v(checksum_variant):
+        # checksum format invalid
+        exit(1)
+
+    metadata_variant = delta.get_child_value(0)
+
+    commit = OSTree.checksum_from_bytes_v(checksum_variant)
+
+    ref = metadata_variant['ref']
+
+    if 'flatpak' in metadata_variant.keys():
+        flatpak = metadata_variant['flatpak']
+    else:
+        flatpak = ''
+
+    if 'origin' in metadata_variant.keys():
+        origin = metadata_variant['origin']
+    else:
+        origin = ''
+
+    if 'runtime-repo' in metadata_variant.keys():
+        runtime_repo = metadata_variant['runtime-repo']
+    else:
+        runtime_repo = ''
+
+    if 'metadata' in metadata_variant.keys():
+        app_metadata = metadata_variant['metadata']
+    else:
+        app_metadata = ''
+
+    if 'gpg-keys' in metadata_variant.keys():
+        gpg_keys = metadata_variant['gpg-keys']
+    else:
+        gpg_keys = ''
+
+    logging.debug("commit: " + commit)
+    logging.debug("flatpak: " + commit)
+    logging.debug("ref: " + ref)
+    logging.debug("origin: " + origin)
+    logging.debug("runtime_repo: " + runtime_repo)
+    logging.debug("app_metadata:\n---\n" + app_metadata + "---")
+    logging.debug("gpg_keys: " + gpg_keys)
+
+    logging.debug("metadata keys: " + str(metadata_variant.keys()))
+
+
+    # open repository
+
+    repo_file = Gio.File.new_for_path(args.repo)
+    repo = OSTree.Repo(path=repo_file)
+    if os.path.exists(os.path.join(args.repo, 'config')):
+        logging.info('Opening repo at ' + args.repo)
+        repo.open()
+    else:
+        logging.info('Creating archive-z2 repo at ' + args.repo)
+        os.makedirs(args.repo)
+        repo.create(OSTree.RepoMode.ARCHIVE_Z2)
+
+
+    # prepare transaction
+
+    repo.prepare_transaction(None)
```

Comment:
You probably would want to wrap this whole thing with `try/except` to abort the transaction if something goes wrong.
```
try:
    repo.prepare_transaction()
    ...
    repo.commit_transaction()
except:
    repo.abort_transaction()
    raise
```